components/automate-dex/web/templates/login.html

```diff
@@ -10,25 +10,23 @@
     <link href="{{ url .ReqPath "static/main.css" }}" rel="stylesheet">
     <link href="{{ url .ReqPath "theme/styles.css" }}" rel="stylesheet">
   </head>
-  <div class="theme-panel">
-    <div class="logo">
-      <img class="theme-navbar__logo" src="{{url .ReqPath "static/img/automate-blue-d9789f4b.svg" }}">
-    </div>
-    <div>
-      {{ range $c := .Connectors }}
-      <div class="theme-form-row">
-        <a href="{{ $c.URL }}" target="_self">
-          <div class="dex-btn theme-btn-provider">
-              <span class="dex-btn-icon dex-btn-icon--{{ $c.ID }}"></span>
-              {{ if eq $c.ID "local" }}
-              <span class="dex-btn-text">Sign in as a local user</span>
-              {{ else }}
-              <span class="dex-btn-text">Sign in with {{ $c.Name }}</span>
-              {{ end }}
-          </div>
-        </a>
+  <body>
+    <main class="theme-panel">
+      <div class="logo">
+        <img class="theme-navbar__logo" alt="Chef Automate" src="{{url .ReqPath "static/img/automate-blue-d9789f4b.svg" }}">
       </div>
-      {{ end }}
-    </div>
-  </div>
+      <h1>User Sign In</h1>
+      <div>
+        {{ range $c := .Connectors }}
+        <form class="theme-form-row" method="post" action="{{ $c.URL }}">
+          {{ if eq $c.ID "local" }}
+          <button type="submit" class="dex-btn-text"><span class="dex-btn-icon dex-btn-icon--{{ $c.ID }}"></span>Sign in as a local user</button>
```

Comment:
:x: This is the problem: `button type="submit"` switches the request for `/password.html` from a GET to a POST. With GET, we'd trigger [this flow in dex](https://github.com/dexidp/dex/blob/2ca992e9b359eff69f9886d10620f255c4cff2f2/server/handlers.go#L330-L333), with `POST`, we'll trigger [that one](https://github.com/dexidp/dex/blob/2ca992e9b359eff69f9886d10620f255c4cff2f2/server/handlers.go#L363-L391).

The latter one expects that it's the entered credentials being `POST`ed.

I'll play with [formmethod](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/submit#formmethod) to see if it gets us out of this predicament.

Reply:
Looks like it might not be possible to retain the query string. I've tried various combinations, like having `<form method="get"` and put the `$c.URL` into the buttons' `formaction`, it will always be overridden.

Some random SO post on the issue: https://stackoverflow.com/a/9882750.

The problem is that the auth request ID is embedded in `$c.URL`, and we don't have access to that; so we can't use it to reconstruct the query string for the GET.

I guess we could come up with some JS to extract it, but maybe there's other a11y-conform ways to get the buttons into a shape where they both are accessible and don't need `type="submit"`? 😦